git_machete/cmd.py

```diff
@@ -339,359 +339,1027 @@ def popen_git(cli_ctxt: CommandLineContext, git_cmd: str, *args: str, **kwargs:
+                second_part = f"Consider using `git machete fork-point --override-to=<revision>|--override-to-inferred|--override-to-parent {yellow_edge_branches[0]}`\n" \
+                              f"or reattaching `{yellow_edge_branches[0]}` under a different parent branch"
+            else:
+                second_part = "Consider using `git machete fork-point --override-to=<revision>|--override-to-inferred|--override-to-parent <branch>` for each affected branch" \
+                              "or reattaching the affected branches under different parent branches"
+
+            sys.stderr.write("\n")
+            warn(f"{first_part}.\n\n{second_part}.")
+
+    def delete_unmanaged(self) -> None:
+        branches_to_delete = excluding(local_branches(self.cli_ctxt), self.managed_branches)
+        cb = current_branch_or_none(self.cli_ctxt)
+        if cb and cb in branches_to_delete:
+            branches_to_delete = excluding(branches_to_delete, [cb])
+            print(fmt(f"Skipping current branch `{cb}`"))
+        if branches_to_delete:
+            branches_merged_to_head = merged_local_branches(self.cli_ctxt)
+
+            branches_to_delete_merged_to_head = [b for b in branches_to_delete if b in branches_merged_to_head]
+            for b in branches_to_delete_merged_to_head:
+                rb = strict_counterpart_for_fetching_of_branch(self.cli_ctxt, b)
+                is_merged_to_remote = is_ancestor_or_equal(self.cli_ctxt, b, rb,
+                                                           later_prefix="refs/remotes/") if rb else True
+                msg_core = f"{bold(b)} (merged to HEAD{'' if is_merged_to_remote else f', but not merged to {rb}'})"
+                msg = f"Delete branch {msg_core}?" + pretty_choices('y', 'N', 'q')
+                opt_yes_msg = f"Deleting branch {msg_core}"
+                ans = ask_if(self.cli_ctxt, msg, opt_yes_msg)
+                if ans in ('y', 'yes'):
+                    run_git(self.cli_ctxt, "branch", "-d" if is_merged_to_remote else "-D", b)
+                elif ans in ('q', 'quit'):
+                    return
 
-    onto: Optional[str] = cli_ctxt.opt_onto
-    if onto:
-        expect_in_managed_branches(onto)
+            branches_to_delete_unmerged_to_head = [b for b in branches_to_delete if b not in branches_merged_to_head]
+            for b in branches_to_delete_unmerged_to_head:
+                msg_core = f"{bold(b)} (unmerged to HEAD)"
+                msg = f"Delete branch {msg_core}?" + pretty_choices('y', 'N', 'q')
+                opt_yes_msg = f"Deleting branch {msg_core}"
+                ans = ask_if(self.cli_ctxt, msg, opt_yes_msg)
+                if ans in ('y', 'yes'):
+                    run_git(self.cli_ctxt, "branch", "-D", b)
+                elif ans in ('q', 'quit'):
+                    return
+        else:
+            print("No branches to delete")
+
+    @staticmethod
+    def edit(cli_ctxt: CommandLineContext) -> int:
```

Comment:
Probably no method of `MacheteClient` whatsoever should have a `CommandLineContext` as a parameter, right? 🤔 all should rely on `self.cli_ctxt` instead

Reply:
moved to instance method